### PR TITLE
<Sidebar/> - show gradient dynamically

### PR DESCRIPTION
**Pull request by miby-wix**
### 🔦 Summary

Sidebar gradient is currently shown accroding to hte component status on mount.
If the status changes (dynamically load more sidebar items or window resize) it's visibility status will not change.
The fix is to make is dynamic to response to both mentioned scenarios.
Added e2e tests as well.

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ V ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/Button/Button.uni.driver.js [hunk 1]
```diff
@@ -25,6 +25,12 @@ export const buttonDriverFactory = base => {
      */
     isButtonDisabled: buttonNextDriver.isButtonDisabled,
 
+    /**
+     * Clicks on the button
+     * @return {Promise<void>}
+     */
+    click: buttonNextDriver.click,
```
Comment (gendeld) [inline src/Button/Button.uni.driver.js:32]: BaseUniDriver already has a click method

File: src/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -62,8 +63,44 @@ class Sidebar extends Component {
     isScrollbarDisplayed: false,
   };
 
-  componentDidMount() {
```
Comment (gendeld) [inline src/Sidebar/Sidebar.js:65]: Are we no longer checking if gradient is needed on mount?

File: src/Sidebar/test/Sidebar.e2e.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { browser } from 'protractor';
```
Comment (gendeld) [inline src/Sidebar/test/Sidebar.e2e.js:1]: I think browser is available without import

File: src/Sidebar/test/Sidebar.e2e.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    const clickAddItemButton = async () => {
+      const buttonDriver = ButtonTestkit({
+        dataHook: storySettings.dataHooks.addItemButton,
+      });
+      await buttonDriver.click();
+    };
+
+    const resizeBoxHeight = () => {
+      return browser.executeScript(
```
Comment (gendeld) [inline src/Sidebar/test/Sidebar.e2e.js:37]: Can't we do this using a button or something instead of direct manipulation?

Comment (miby-wix) [reply src/Sidebar/test/Sidebar.e2e.js:37]: I investigated the protractor capabilities and couldn't find any other way to manipulate the element (I can get the size easily but no change it...)
Do you know another way of how to do so?

Comment (gendeld) [reply src/Sidebar/test/Sidebar.e2e.js:37]:
```
browser.driver.manage().window().setSize(width, height);
```

Comment (miby-wix) [reply src/Sidebar/test/Sidebar.e2e.js:36]: that's for the window... not the specific component...

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6471

File: src/Sidebar/test/Sidebar.e2e.js (hunk 4 above)
Comment (gendeld) [reply src/Sidebar/test/Sidebar.e2e.js:36]: Then just have a button that changes the component's height state maybe?

Comment (gendeld) [reply src/Sidebar/test/Sidebar.e2e.js:36]: https://preview.wix-style-react.com/pr_6471/?path=/story/introduction-playground--playground&snippet=9ef8cb99-a4d8-4dfe-ab91-5992a7a5a326

Comment (miby-wix) [reply src/Sidebar/test/Sidebar.e2e.js]: like!

File: src/Sidebar/test/Sidebar.e2e.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      expect(driver.isGradientDisplayed()).toBe(false);
+    });
+
+    it('Should show gradient when items list is larger than list container', async () => {
+      await clickAddItemButton();
+      expect(driver.isGradientDisplayed()).toBe(true);
+    });
+
+    it('Should show gradient when container is resized to be smaller than items list', async () => {
+      clickSetHeightButton();
```
Comment (gendeld) [inline src/Sidebar/test/Sidebar.e2e.js:52]:
```suggestion
      expect(driver.isGradientDisplayed()).toBe(false);
      await clickSetHeightButton();
```

**Review by gendeld — APPROVED** [review]

**Pull request merged**